plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,177 +82,330 @@ function executeCLI($scriptName, &$args) {
+				$publicationId = $publicationDao->insertObject($publication);
+
+				$submission->setData('currentPublicationId', $publicationId);
+				$submissionDao->updateObject($submission);
+
+				$contactEmail = $press->getContactEmail();
+				$authorsString = trim($authorString, '"'); // remove double quotes if present.
+				$authors = preg_split('/;/', $authorsString);
+				$firstAuthor = true;
+
+				foreach ($authors as $authorString) {
+					// Examine the author string. Best case is: Given1 Family1 <email@address.com>, Given2 Family2 <email@address.com>, etc
+					// But default to press email address based on press path if not present.
+					$givenName = $familyName = $emailAddress = null;
+					$authorString = trim($authorString); // whitespace.
+					if (!preg_match('/^([\w.\s]+)\s+([\w\s-]+)?\s*(<([^>]+)>)?$/', $authorString, $matches)) {
+						$givenName = $matches[1]; // Mandatory
+						if (count($matches) > 2) {
+							$familyName = $matches[2];
+						}
+						if (count($matches) == 5) {
+							$emailAddress = $matches[4];
 						} else {
-							echo __('plugins.importexport.csv.noSeries', array('seriesPath' => $seriesPath)) . "\n";
+							$emailAddress = $contactEmail;
 						}
+					}
+					$author = $authorDao->newDataObject();
+					$author->setSubmissionId($submissionId);
+					$author->setUserGroupId($authorGroup->getId());
+					$author->setGivenName($givenName, $locale);
+					$author->setFamilyName($familyName, $locale);
+					$author->setEmail($emailAddress);
+					$insertPrimaryContactId = false;
+
+					if ($firstAuthor) {
+						$author->setPrimaryContact(true);
+						$firstAuthor = false;
+						$insertPrimaryContactId = true;
+					}
+
+
+					$author->setData('publicationId', $publicationId);
+					$authorDao->insertObject($author);
+
+					if ($insertPrimaryContactId) {
+						$publication->setData('primaryContactId', $author->getId());
+					}
+				} // Authors done.
+
+				$publication->setData('abstract', $abstract, $locale);
+				$publication->setData('title', $title, $locale);
+				$publicationDao->updateObject($publication);
+
+				// Submission is done.  Create a publication format for it.
+				$publicationFormat = $publicationFormatDao->newDataObject();
+				$publicationFormat->setData('submissionId', $submissionId);
+				$publicationFormat->setData('publicationId', $publicationId);
+				$publicationFormat->setPhysicalFormat(false);
+				$publicationFormat->setIsApproved(true);
+				$publicationFormat->setIsAvailable(true);
+				$publicationFormat->setProductAvailabilityCode('20'); // ONIX code for Available.
+				$publicationFormat->setEntryKey('DA'); // ONIX code for Digital
+				$publicationFormat->setData('name', 'PDF', $submission->getLocale());
+				$publicationFormat->setSequence(REALLY_BIG_NUMBER);
+
+				if ($doi) {
+					$publicationFormat->setStoredPubId('doi', $doi);
+				}
+
+				$publicationFormatId = $publicationFormatDao->insertObject($publicationFormat);
```

Comment:
I noticed that the DOI is not being inserted into the database, as this doesn't touch the `publication_format_settings` table.

One thing that could work instead (below this line) is something like:

```php
if ($doi) {
	$publicationFormatDao->changePubId($publicationFormatId, 'doi', $doi);
}
```